docs/project/contribution_tools.md

```diff
@@ -71,6 +72,16 @@ pre-commit install
 ./scripts/run_bazelisk.py test //...:all
 ```
 
+### Configuring `clangd`
+
+To ensure that `clangd` reports accurate diagnostics it
+needs a generated file called `compile_commands.json`, this can be generated by
+invoking the command below.
+
+```
+./scripts/create_compdb.py
+```
+
```

Comment:
We've been keeping "Setup" focused on things that are necessary to build and contribute to Carbon. Perhaps this should go under the "Tools" section instead, near "Optional tools"? We already have a mention of vscode setup there, and I think clangd is in a similar category for IDE support.